wallet/address.js

```diff
@@ -79,24 +79,24 @@ function isValidAddress(address) {
   return true
 }
 
-async function deriveAddressAndSecret(rootSecretString, childIndex) {
-  let addressPayload, addressAttributes, derivedSecretString, addressRoot
+async function deriveAddressWithHdNode(parentHdNode, childIndex) {
+  let addressPayload, addressAttributes, derivedHdNode, addressRoot
 
   if (childIndex === 0x80000000) {
     // root address
     addressPayload = new Buffer(0)
     addressAttributes = new Map()
-    derivedSecretString = rootSecretString
+    derivedHdNode = parentHdNode
   } else {
     // the remaining addresses
-    const hdPassphrase = await deriveHDPassphrase(rootSecretString)
+    const hdPassphrase = await deriveHdPassphrase(parentHdNode)
     const derivationPath = [0x80000000, childIndex]
 
     addressPayload = encryptDerivationPath(derivationPath, hdPassphrase)
     addressAttributes = new Map([[1, cbor.encode(addressPayload)]])
-    derivedSecretString = deriveSK(rootSecretString, childIndex)
+    derivedHdNode = deriveSk(parentHdNode, childIndex)
```

Comment:
I guess we can rename `deriveSk` -> `deriveHdNode`

Reply:
I've noticed, makes sense, but then also `deriveSkIteration -> deriveHdNodeIteration`